javascript/signin.js

```diff
@@ -0,0 +1,122 @@
+      emailInput.parentNode.removeChild(errorTextEmail);
+    }
+    // 이메일 미입력 에러
+    if (email == '') {
+      showError(errorTextEmail, '이메일을 입력해 주세요.', emailInput);
+      return false;
+    }
+    // 이메일 형식 확인
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!emailRegex.test(email)) {
+      showError(errorTextEmail, '올바른 이메일 주소가 아닙니다.', emailInput);
+      return false;
+    }
+    return true;
+  }
+
+  // 비번 확인 함수
+  function passwordCheck() {
+    const password = passwordInput.value.trim();
+    clearErrorStyle(passwordInput);
+    // 기존 에러 메세지 삭제
+    if (errorTextPassword.parentNode === passwordInput.parentNode) {
+      passwordInput.parentNode.removeChild(errorTextPassword);
+    }
+    // 비밀번호 미입력 에러
+    if (password == '') {
+      showError(errorTextPassword, '비밀번호를 입력해 주세요.', passwordInput);
+      return false;
+    }
+    return true;
+  }
+
+  // 에러 스타일 빼기
+  function clearErrorStyle(inputElement) {
+    const parentContainer = inputElement.parentNode;
+    const errorElement = parentContainer.querySelector('.error-message');
+
+    parentContainer.classList.remove('error');
+    if (errorElement == true) {
+      parentContainer.removeChild(errorTextEmail);
+    }
+  }
+
+  // 에러 메세지 띄우기
+  function showError(errorText, message, inputElement) {
+    errorText.textContent = message;
+    errorText.classList.add('error-message');
+    inputElement.parentNode.appendChild(errorText);
+    inputElement.parentNode.classList.add('error');
+  }
+
+  // 눈 아이콘 변경하기
+  function togglePassword() {
+    const coverEye = '../image/icon/eye_closed.svg';
+    const openEye = '../image/icon/eye_open.svg';
+    if (isPasswordCovered) {
+      passwordCover.src = coverEye;
+    } else {
+      passwordCover.src = openEye;
+    }
+    isPasswordCovered = !isPasswordCovered;
+  }
+
+  function showPassword() {
+    if (isPasswordCovered) {
+      passwordInput.type = 'password';
+    } else {
+      passwordInput.type = 'text';
+    }
+  }
+
+  //////////////// 함수 사용////////////////////
+
+  // 이메일 이벤트 리스너 부여
+  emailInput.addEventListener('focusout', function () {
+    emailCheck();
+  });
+
+  // 비번 이벤트 리스너 부여
+  passwordInput.addEventListener('focusout', function () {
+    passwordCheck();
+  });
+
+  // 로그인 버튼 입력 시 이메일 비번 확인 후 이동
+  form.addEventListener('submit', function (event) {
+    const codeit = { email: 'test@codeit.com', password: 'codeit101' };
+    const email = emailInput.value.trim();
+    const password = passwordInput.value.trim();
+
+    if (email === codeit.email && password === codeit.password) {
+      event.preventDefault();
+      window.location.href = '../html/folder.html';
+    } else {
+      showError(errorTextEmail, '이메일을 확인해 주세요.', emailInput);
+      showError(errorTextPassword, '비밀번호를 확인해 주세요.', passwordInput);
+      event.preventDefault();
+    }
+  });
+
+  passwordCover.addEventListener('click', function () {
+    showPassword();
+    togglePassword();
+  });

```

Comment:
이렇게 이벤트 핸들러 내에서 여러 함수를 호출해야 하는 경우에는 따로 핸들러 함수로 분리해보면 어떨까요?

```suggestion
 const handlePasswordClick = ()=> { 
    showPassword();
    togglePassword();
 } 
  passwordCover.addEventListener('click', handlePasswordClick);
```

위와 같이 핸들러 함수를 만들어주고, 이벤트 리스너에 할당하면 일일이 필요한 함수들을 명령해주는 것 보다 더 가독성이 좋을 것 같아요. 
추후에 해당 이벤트가 트리거 되었을 때 호출되어야 할 함수가 더 추가될 때도 이벤트리스너까지 갈 필요 없이 핸들러함수만 수정해주면 될 것 같구요 ㅎㅎ
다른 이벤트리스너들의 콜백함수들도 마찬가지로 적용되면 좋을 것 같아요~

Reply:
추후 추가되는 함수들도 고려해보면 해두면 참 좋을 것 같은 부분이네요👍 수정했습니다! 8e4fcd0ecf68a24e453f451ab67114d9672fe766